kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/food2door/AppConfig.java

```diff
@@ -0,0 +1,81 @@
+package com.kodilla.good.patterns.food2door;
+
+import com.kodilla.good.patterns.food2door.data.Company;
+import com.kodilla.good.patterns.food2door.demands.*;
+import com.kodilla.good.patterns.food2door.process.*;
+import com.kodilla.good.patterns.food2door.services.*;
+
+import java.util.*;
+
+// dummy implementation of config class. Normally it should read data from config file, for example config.xml
+public final class AppConfig {
+    private Company appUser;
+    private DemandRetriever demandRetriever;
+    private  StorageService storageService;
+    private  Map<Company, OfferService> offerServices;
+    private  Map<Company, OrderService> orderServices;
+    private OrderProcessorResultHandler orderProcessorResultHandler;
+
+    public AppConfig(String configFile) {
+        // demo
+        parseConfigFile(configFile);
+    }
+
+    private boolean parseConfigFile(String configFile) {
+        // dummy implementation
+
+        appUser = new Company(1, "Food2Door", "ul. Handlowa 10, 12-345 Fudowo", "food2door@example.com", "123456789");
+        storageService = new MySqlStorageService("jdbc:mysql://localhost:3306/storage", "supply_processor",
+                                                 "IUyhgTy^599*JH");
+        demandRetriever = new DefaultDemandRetriever();
+
+        Company efs = new Company(101, "ExtraFoodShop", "ul. Rolnicza 1, 11-111 Kapustowo", "efs@example.com",
```

Comment:
Let assume that I want to add new company and their offer - it means that I have to modify this method - this violate Open/Closed principle.

Reply:
Nobody will add new company inside a dummy parseeConfigFile(String configFile). Companies' data are beeing written in a config file and simultaneously an adequate implementation of their offer and order services are coded. I simply skip a support of the config file.  So then I  provided only a part of the solution. So I must correct it.